feat(issue-owners): Add to Code Owners and eval flow (#4560)
getsentry/sentry-docs: src/docs/product/issues/issue-owners/index.mdx

description: "Learn how to automatically assign issues to their respective owners, or alert the owners about an issue, allowing you to find the developer with the most context about a fix."
---

> Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with Code Owners and Ownership Rules. Code Owners lets you import your CODEOWNERS file and we will assign according to those filepaths. Ownership Rules allows you to override the assignments from Code Owners and provide advanced matcher types (ex: urls, tags) and can match on file paths of files in the stack trace, URL of the request, or event tags.

Reviewer: Can we link to the Codeowners documentation on github?

> Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with Code Owners and Ownership Rules. Code Owners lets you import your CODEOWNERS file and we will assign according to those filepaths. Ownership Rules allows you to override the assignments from Code Owners and provide advanced matcher types (ex: urls, tags) and can match on file paths of files in the stack trace, URL of the request, or event tags.

Author: @imatwawana the features are called Code Owners and Ownership Rules. I don't think it makes sense to have them lowercase.

> Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with Code Owners and Ownership Rules. Code Owners lets you import your CODEOWNERS file and we will assign according to those filepaths. Ownership Rules allows you to override the assignments from Code Owners and provide advanced matcher types (ex: urls, tags) and can match on file paths of files in the stack trace, URL of the request, or event tags.

Reviewer 2: We have _a lot_ of features so capitalizing all of them becomes cumbersome and when everything is capitalized, the importance of capitalization becomes diluted. Also, the capitalization across docs tends to be inconsistent, so I'm trying to impose a bit more structure in general around when we use it. As a style rule, we capitalize at the page level (so in this case Issue Owners page), but not features that are within a page unless we're doing a direct reference to them in the UI (in which case they'd be in quotation marks, e.g., the "Ownership Rules" section).

> Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with _code owners_ and _ownership rules_. Code owners lets you import your GitHub or GitLab CODEOWNERS file, and then we assign issues according to those file paths. Ownership rules allow you to override the assignments from code owners and provide advanced matcher types (for example, urls and tags). These rules can also match on the file paths of files in the stack trace, URL of the request, or event tags.

Reviewer 2: Fixing a couple grammar issue here that arise with _code owners_ being plural. 

```suggestion
Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with _code owners_ and _ownership rules_. Code owners functionality lets you import your GitHub or GitLab CODEOWNERS file, and then we assign issues according to those file paths. Ownership rules allow you to override the assignments based on code owners and provide advanced matcher types (for example, urls and tags). These rules can also match on the file paths of files in the stack trace, URL of the request, or event tags.
```

> Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with _code owners_ and _ownership rules_. Code owners functionality lets you import your GitHub or GitLab CODEOWNERS file, and then we assign issues according to those file paths. Ownership rules allow you to override the assignments based on code owners and provide advanced matcher types (for example, urls and tags). These rules can also match on the file paths of files in the stack trace, URL of the request, or event tags.

Reviewer 2: ```suggestion
Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with _code owners_ and _ownership rules_. Code owners functionality lets you import your [GitHub or GitLab CODEOWNERS file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners), and then we assign issues according to those file paths. Ownership rules allow you to override the assignments based on code owners and provide advanced matcher types (for example, urls and tags). These rules can also match on the file paths of files in the stack trace, URL of the request, or event tags.
```

> Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with _code owners_ and _ownership rules_. Code owners functionality lets you import your GitHub or GitLab CODEOWNERS file, and then we assign issues according to those file paths. Ownership rules allow you to override the assignments based on code owners and provide advanced matcher types (for example, urls and tags). These rules can also match on the file paths of files in the stack trace, URL of the request, or event tags.

Reviewer 2: ```suggestion
Sentry offers multiple ways to define the "ownership" of an issue. With ownership defined, we can automatically assign issues and send alerts to the owner. Sentry defines ownership with _code owners_ and _ownership rules_. Code owners functionality lets you import your [GitHub](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) or [GitLab](https://docs.gitlab.com/ee/user/project/code_owners.html) CODEOWNERS file, and then we assign issues according to those file paths. Ownership rules allow you to override the assignments based on code owners and provide advanced matcher types (for example, urls and tags). These rules can also match on the file paths of files in the stack trace, URL of the request, or event tags.
```
